ir/memory.cpp

```diff
@@ -218,17 +223,27 @@ pair<expr, vector<expr>> Memory::mkInput(const char *name) {
   return { Pointer(*this, offset, local_bid, bid).release(), { var } };
 }
 
-expr Memory::alloc(const expr &bytes, unsigned align, bool local) {
+expr Memory::alloc(const expr &bytes, unsigned align, bool local, bool heap) {
```

Comment:
My idea with the `local` flag was to mark is as stack or heap. Maybe just get rid of `local` altogether?

Reply:
Ok. I can rename `Pointer::get_local_bid()` / `Pointer::get_nonlocal_bid()` etc if you want.

Reply:
My idea was that local blocks ids would be just for stack, while non-local would be for globals + heap. This is because when we get a pointer as argument it can be either in globals or heap. (it can also be in the stack of some other function, but definitely not on the stack of the current function).
This idea was a cheap way to do ensure that stuff on stack doesn't alias anything else. We don't need any constraint; it's already given by the separated block ids.

Hence I would like to keep the concepts local/non-local as something different from heap/global/stack allocation.